packages/ui/src/components/Navigation/Navigation.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+    ArrowUpRightIcon,
+    Bars3BottomRightIcon,
+    MagnifyingGlassIcon,
+    XMarkIcon,
+} from '@heroicons/react/24/outline';
+import type {
+    Category,
+    Culture,
+    ExtendedStory,
+    NewsroomLanguageSettings,
+    UploadedImage,
+} from '@prezly/sdk';
+import Image from '@prezly/uploadcare-image';
+import Link from 'next/link';
+import { type MouseEvent as ReactMouseEvent, useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+import { useDevice } from '@/hooks';
+
+import { Button } from '../Button';
+
+import { CategoriesDropdown, LanguagesDropdown } from './components';
+import type { NavigationLayout } from './types';
+import { extractDomainFromUrl } from './util';
+
+export interface Props {
+    className?: string;
+    categories: Category[];
+    languages: NewsroomLanguageSettings[];
+    layout?: NavigationLayout;
+    logo?: UploadedImage | null;
+    siteName: string;
+    publicGalleriesCount: number;
+    externalSiteLink?: string;
+    isSearchEnabled?: boolean;
+    onSearch?: () => void;
+    locale: Culture['code'];
+    hasStandaloneAboutPage?: boolean;
+    hasStandaloneContactsPage?: boolean;
+    hasError?: boolean;
+    currentStory?: ExtendedStory;
+    currentCategory?: Pick<Category, 'i18n' | 'display_name'>;
```

Comment:
These props should not be optional, since they're required for Language Picker to work properly.
```suggestion
    currentStory: ExtendedStory | undefined;
    currentCategory: Pick<Category, 'i18n' | 'display_name'> | undefined;
```